### PR TITLE
Initial MS CNG work

### PR DESCRIPTION
**Pull request by vmiklos**
Two commits:

- first is just a skeleton + the nmake and javascript bits
- second is sha256 support

This is already quite some code to read through, so I stopped here, but I plan the ecdsa-sha256 transform as the next step.

### CODE REVIEW

File: include/xmlsec/mscng/app.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
+/*
+ * XML Security Library (http://www.aleksey.com/xmlsec).
+ *
+ * This is free software; see Copyright file in the source
+ * distribution for preciese wording.
+ *
+ * Copyright (C) 2002-2016 Aleksey Sanin <aleksey@aleksey.com>. All Rights Reserved.
```
Comment (lsh123) [inline include/xmlsec/mscng/app.h]: You should change it :)

File: src/mscng/crypto.c [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    gXmlSecMSCngFunctions->cryptoInit                   = xmlSecMSCngInit;
+    gXmlSecMSCngFunctions->cryptoShutdown               = xmlSecMSCngShutdown;
+    gXmlSecMSCngFunctions->cryptoKeysMngrInit           = xmlSecMSCngKeysMngrInit;
+
+    /********************************************************************
+     *
+     * Key data ids
+     *
+     ********************************************************************/
+#if 0 /* TODO */
```
Comment (lsh123) [inline src/mscng/crypto.c]: nitpick: I would use #ifdef XMLSEC_MSCNG_TODO (or something like it) so it is easier to find it later.

File: src/mscng/digests.c [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+    xmlSecAssert2(ctx != NULL, -1);
+
+    if(transform->status == xmlSecTransformStatusNone) {
+        /* open an algorithm handle */
+        ret = BCryptOpenAlgorithmProvider(
+            &ctx->hAlg,
+            ctx->pszAlgId,
+            NULL,
+            0);
+
```
Comment (lsh123) [inline src/mscng/digests.c]: remove empty line?

File: include/xmlsec/mscng/app.h (hunk 1 above)
Comment (vmiklos) [reply include/xmlsec/mscng/app.h]: Done.

File: src/mscng/crypto.c (hunk 2 above)
Comment (vmiklos) [reply src/mscng/crypto.c]: I now did that, though I left the original comment at the #endif part, to help knowing what would be the proper #if once the TODO can be removed.

File: src/mscng/digests.c (hunk 3 above)
Comment (vmiklos) [reply src/mscng/digests.c]: Right, that was not intentional, I left no empty line between the crypto call and the error handling at other places.

**Pull request merged**